### PR TITLE
Simplify config create

### LINKED ISSUE #304: Simplify 'config create' step
Since for a default Leap user we require only API token, and all other prompts may confuse some (i.e. config path, profile name, endpoint, solver, client), we might consider splitting the "config create" flow in two branches: (1) default/basic; and (2) advanced.

In the advanced branch we can then also add a few more non-essential parameter prompts (e.g. timeout values).

Seems like one config issue users reported so far is related to config file path (if user changes the path while not understanding the consequences). Implicitly using a default there (in basic branch) will help. Use of the default (undefined) solver is also strongly encouraged (because undefined solver is translated to `{"online": True}`. That addressed the problem of hard-coding solver names too (as older versions of client did).

### PR DESCRIPTION
**Pull request by randomir**
### New default (minimal config flow)
```
$ dwave config create
Using the simplified configuration flow.
Try 'dwave config create --full' for more options.

Creating new configuration file: /home/user/.config/dwave/dwave.conf
Profile [defaults]: ⏎
Updating existing profile: defaults
Authentication token [skip]: TOK-en⏎
Configuration saved.
```

```
$ cat /home/user/.config/dwave/dwave.conf
[defaults]
token = TOK-en
```

### "Full" config (old behavior)
```
$ dwave config create --full
Updating existing configuration file: /home/user/.config/dwave/dwave.conf
Confirm configuration file path [/home/user/.config/dwave/dwave.conf]: ⏎
Available profiles: defaults
Profile (select existing or create new) [defaults]: prod⏎
Creating new profile: prod
Solver API endpoint URL [skip]: https://endpoint.com⏎
Authentication token [TOK-en]: DEV-xx⏎
Client class [skip]: ⏎
Solver [skip]: {"num_active_qubits__gt": 4000}⏎
Configuration saved.
```
```
$ cat /home/user/.config/dwave/dwave.conf
[defaults]
token = TOK-en

[prod]
endpoint = https://endpoint.com
token = DEV-xx
solver = {"num_active_qubits__gt": 4000}
```

Close #304.

### CODE REVIEW

File: dwave/cloud/cli.py [hunk 1]
```diff
@@ -82,23 +85,28 @@ def _print_deprecation(ctx, param, value, msg=None, update=None):
...
-        help='Configuration file path')(fn)
-    fn = click.option(
-        '-c', default=None, expose_value=False,
-        type=click.Path(exists=True, dir_okay=False),
-        help="[Deprecated in favor of '-f']",
-        callback=deprecated_option(CONFIG_FILE_DEPRECATION_MSG, update='config_file'))(fn)
-    fn = click.option(
-        '--profile', '-p', default=None,
-        help='Connection profile (section) name')(fn)
+    Optionally skip existency check by click with `exist=False`.
```
Comment (JoelPasvolsky) [inline dwave/cloud/cli.py:91]: `exists=False`

File: dwave/cloud/cli.py [hunk 2]
```diff
@@ -175,84 +183,118 @@ def inspect(config_file, profile):
...
 
 @config.command()
-@config_file_options
-def create(config_file, profile):
-    """Create and/or update cloud client configuration file."""
-    return _config_create(config_file, profile)
-
+@config_file_options(exists=False)
+@click.option('--full', 'ask_full', is_flag=True,
+              help='Configure non-essential options (like endpoint and solver).')
```
Comment (JoelPasvolsky) [inline dwave/cloud/cli.py:188]: (like endpoint and solver)  --> (such as endpoint and solver)

**Review by JoelPasvolsky — APPROVED** [review]
LGTM. 
* Does the user get the simplified flow also when using `dwave setup`?
* You can now specify a config file anywhere with any name (e.g., `/tmp/x`) -- so how does `dwave config ls -m` find such files?
* Need to update https://docs.ocean.dwavesys.com/en/latest/docs_cli.html

**Comment by randomir** [comment]
Thanks for reviewing, @JoelPasvolsky.

Answers to your questions:
- `dwave setup` will use the new default / simplified flow
- `dwave config ls -m` will not find files outside of predefined OS-specific locations. The ability to specify config file path existed before, but the command failed for nonexistent files (a bug now fixed).
- yes, we should update the SDK docs, good catch.

File: dwave/cloud/cli.py [hunk 3]
```diff
@@ -22,6 +22,9 @@
 from functools import partial
 from timeit import default_timer as timer
```
Comment (arcondello) [inline dwave/cloud/cli.py:24]: should there be a blank line here? Could I ask a more nit-picky question?

**Review by arcondello — APPROVED** [review]
LGTM

**Pull request merged**